pipelines/api-pipelines.yml

```diff
@@ -20,137 +22,57 @@ pr:
-              artifactName: 'drop'
-              publishLocation: 'Container'
-
-  - stage: DeployStage
-    dependsOn: BuildStage
+  - stage: Development
+    displayName: Development
+    dependsOn: BuildAndValidate
     condition: and(succeeded(), ne(variables['Build.Reason'], 'PullRequest'))
-    displayName: Deploy
+    variables:
+      - group: prodoh-urlist-dev
     jobs:
-      - job: RegionalMaxtrixDeployment
-        displayName: API to Region
-        strategy:
-          matrix:
-            westus2:
-              region: westus2
-            eastus:
-              region: eastus
-            centralus:
-              region: centralus
-        steps:
-          - task: UseDotNet@2
-            displayName: Dotnet version 3.1.100
-            inputs:
-              version: '3.1.100'
+      - template: templates/api-deploy-environment.yml
+        parameters:
+          subscription: "PELITTLE TEAM - CSE DWR(d36d0808-a967-4f73-9fdc-32ea232fc81d)"
+          businessUnit: $(businessUnit)
+          appName: $(appName)
+          environment: dev
+          regions:
+            - westus2
 
-          - task: DownloadBuildArtifacts@0
-            displayName: Download Build Artifacts
-            inputs:
-              buildType: 'current'
-              downloadType: 'single'
-              artifactName: 'drop'
-              downloadPath: '$(Build.ArtifactStagingDirectory)'
-
-          - task: AzureRMWebAppDeployment@4
-            displayName: Deploy to Staging Slot
-            inputs:
-              ConnectionType: 'AzureRM'
-              azureSubscription: $(azureSubscription)
-              appType: 'apiApp'
-              ResourceGroupName: 'rg-$(businessUnit)-$(appName)-$(environment)-$(region)'
-              WebAppName: 'backend-$(businessUnit)-$(appName)-$(environment)-$(region)'
-              deployToSlotOrASE: true
-              SlotName: staging
-              packageForLinux: '$(Build.ArtifactStagingDirectory)/**/*.zip'  
-              appSettings: 
-                -CosmosSettings:ServiceEndpoint $(COSMOSDB_ENDPOINT) 
-                -CosmosSettings:AuthKey $(COSMOSDB_AUTHKEY) 
-                -CosmosSettings:DatabaseName linkylinkdb 
-                -CosmosSettings:ContainerName linkbundles
-                -AzureAdB2C:Instance $(AZUREADB2C_INSTANCE)
-                -AzureAdB2C:ClientId $(AZUREADB2C_CLIENTID)
-                -AzureAdB2C:Domain $(AZUREADB2C_DOMAIN)
-                -AzureAdB2C:SignUpSignInPolicyId $(AZUREADB2C_POLICYID)
+  - stage: Staging
+    displayName: Staging
+    dependsOn: Development
+    variables:
+      - group: prodoh-urlist-staging
+    jobs:
+      - template: templates/api-deploy-environment.yml
+        parameters:
+          subscription: "PELITTLE TEAM - CSE DWR(d36d0808-a967-4f73-9fdc-32ea232fc81d)"
+          businessUnit: $(businessUnit)
+          appName: $(appName)
+          environment: staging
+          regions:
+            - westus2
+            - eastus
```

Comment:
When we swap with prod, do we swap for all staging regions at the same time? if so, given its staging, can we minimize area of impact and only deploy to a single region first?

Reply:
Right now the regions are deployed in a matrix (parallel) to reduce time.  So yes, whey would all swap around the same time.  We can choose to make this synchronous if needed.  Let's discuss.